### PR TITLE
[4.0] ES6 please show on

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- script converted to vanilla
- showon is using class 

### Testing Instructions
Check the administrator global configuration. Things should appear/disappear depending on some switches

### Expected result
screenshot 2018-08-12 at 18 24 08
screenshot 2018-08-12 at 18 24 18
screenshot 2018-08-12 at 18 24 27

### CODE REVIEW

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on 5be2644849d9d8ef172c43dc52a1997dc7247df6<br /><br />After applying the patch, the showon buttons continue to work as before in Global Configuration and plugins<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21568">issues.joomla.org/tracker/joomla-cms/21568</a>.</sub>

File: build/media_src/system/js/showon.es6.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
+/**
+ * @copyright  Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+// @todo remove this totally irrelevant piece of code from this file
```
Comment (roland-d) [inline build/media_src/system/js/showon.es6.js:6]: Why is this todo still here?

File: build/media_src/system/js/showon.es6.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    /*
+     * Constructor
+     *
+     * @param {HTMLElement} cont Container element
+     */
+    constructor(cont) {
+      const self = this;
+      this.container = cont || document;
+      this.fields = {
+        // origin-field-name: {
```
Comment (roland-d) [inline build/media_src/system/js/showon.es6.js:36]: Do we need the commented out code?

File: build/media_src/system/js/showon.es6.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+          condition.valid = 0;
+
+          // Test in each of the elements in the field array if condition is valid
+          this.fields[key].origin.forEach((originField) => {
+            if (originField.name !== elementShowonData.field) {
+              return;
+            }
+
+            const originId = originField.id;
+
```
Comment (roland-d) [inline build/media_src/system/js/showon.es6.js:144]: An empty line too many

File: build/media_src/system/js/showon.es6.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        // origin-field-name: {
+        //   origin:  ['collection of all the trigger nodes'],
+        //   targets: ['collection of nodes to be controlled control']
+        // }
+      };
+
+      this.showonFields = [].slice.call(this.container.querySelectorAll('[data-showon]'));
+      // Populate the fields data
+      if (this.showonFields.length) {
+        // @todo refactor this, dry
```
Comment (roland-d) [inline build/media_src/system/js/showon.es6.js:45]: Is this comment still valid?

File: build/media_src/system/js/showon.es6.js (hunk 1 above)
Comment (dgrammatiko) [reply build/media_src/system/js/showon.es6.js:6]: this code needs to be transferred to the toolbar.js but in another PR...

File: build/media_src/system/js/showon.es6.js (hunk 2 above)
Comment (dgrammatiko) [reply build/media_src/system/js/showon.es6.js:36]: Is just there so anyone that needs to debug this script knows what the object contains

File: build/media_src/system/js/showon.es6.js (hunk 4 above)
Comment (dgrammatiko) [reply build/media_src/system/js/showon.es6.js:45]: yes, the procedure we use to build the object can be simplified, but it needs some time and I prefer to do the heavy lifting first, there quite some more scripts that we need to move to ES6. (seems that the list is neverending...)

**Pull request merged**